packages/app/src/specs/RunStatusDots.vue

```diff
@@ -0,0 +1,204 @@
+        tabindex="0"
+      >
+        <div
+          v-for="(dot,i) in dotClasses"
+          :key="i"
+          class="ml-4px"
+        >
+          <i-cy-dot-solid_x4
+            width="4"
+            height="4"
+            :class="dot"
+            :data-cy="'run-status-dot-'+i"
+          />
+        </div>
+        <div>
+          <component
+            :is="latestStatus.icon"
+            width="16"
+            height="16"
+            :class="{'animate-spin': latestStatus.spin}"
+            :data-cy="'run-status-dot-latest'"
+            class="ml-4px"
+          />
+        </div>
+      </div>
+    </component>
+    <template
+      #popper
+    >
+      <ExternalLink
+        v-if="latestRun"
+        :href="latestRun.url ?? '#'"
+        :use-default-hocus="false"
+      >
+        <SpecRunSummary
+          :run="latestRun"
+          :spec-file-no-extension="props.specFileName"
+          :spec-file-extension="props.specFileExtension"
+        />
+      </ExternalLink>
+    </template>
+  </component>
+</template>
+
+<script setup lang="ts">
+
+import ExternalLink from '@cy/gql-components/ExternalLink.vue'
+import type { RunStatusDotsFragment } from '../generated/graphql'
+import Tooltip from '@packages/frontend-shared/src/components/Tooltip.vue'
+import { computed } from 'vue'
+import CancelledIcon from '~icons/cy/cancelled-solid_x16.svg'
+import ErroredIcon from '~icons/cy/errored-solid_x16.svg'
+import FailedIcon from '~icons/cy/failed-solid_x16.svg'
+import PassedIcon from '~icons/cy/passed-solid_x16.svg'
+import PlaceholderIcon from '~icons/cy/placeholder-solid_x16.svg'
+import RunningIcon from '~icons/cy/running-outline_x16.svg'
+import SpecRunSummary from './SpecRunSummary.vue'
+import { gql } from '@urql/vue'
+
+gql`
+fragment RunStatusDots on RemoteFetchableCloudProjectSpecResult {
+  id
+  data {
+    ... on CloudProjectSpecNotFound {
+      retrievedAt
+    }
+    ... on CloudProjectSpec {
+      id
+      retrievedAt
+      specRuns(first: 4, fromBranch: $fromBranch) {
+        nodes {
+          id
+          runNumber
+          testsFailed{
+            min
+            max
+          }
+          testsPassed{
+            min
+            max
+          }
+          testsPending{
+            min
+            max
+          }
+          testsSkipped{
+            min
+            max
+          }
+          createdAt
+          groupCount
+          specDuration{
+            min
+            max
+          }
+          status
+          url
+        }
+      }
+    }
+  }
+}
+`
+
+const props = defineProps<{
+  gql: RunStatusDotsFragment | null
+  specFileName: string
+  specFileExtension: string
+}>()
+
+const runs = computed(() => {
+  return props.gql?.data?.__typename === 'CloudProjectSpec' ? props.gql?.data?.specRuns?.nodes ?? [] : []
+})
+
+const dotClasses = computed(() => {
+  const statuses = ['placeholder', 'placeholder', 'placeholder']
+
+  if (runs.value && runs.value.length > 0) {
```

Comment:
I think everything ends up working out, but we could check to see if this is greater than 1 instead of zero, since we start our indexing at 1.

Reply:
We do want this condition to pass if `runs.value.length===1`. We can change the condition to >=1 id that makes it more readable. My personal preference is `>0`, but I'm fine with changing it if that makes it more readable.

Reply:
A length of 1 would result in a no-op for the for-loop, right? Which is fine functionally, but it seems like we could just skip it in that case.